src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -119,7 +120,25 @@ irep_idt generate_java_generic_typet::build_generic_tag(
     INVARIANT(
       is_java_generic_inst_parameter(param),
       "Only create full concretized generic types");
-    new_tag_buffer << param.subtype().get(ID_identifier);
+    const std::string &id(id2string(param.subtype().get(ID_identifier)));
+    new_tag_buffer << id;
+    if(has_prefix(id, "java::array["))
+    {
+      namespacet ns(symbol_table);
```

Comment:
I don't think this `namespacet` is used in this block?

Reply:
yes, this was left over from an earlier approach, gone now